source/plugin/blocks/accessing.rst

```diff
@@ -34,56 +34,104 @@ referenced block is any kind of banner by checking the blocks type.
+        BlockType type = blockLoc.getBlock().getType();
+        return type.equals(BlockTypes.STANDING_BANNER)
+                || type.equals(BlockTypes.WALL_BANNER);
     }
 
 .. tip ::
+    
+    The function ``==`` could be used in place of ``equals()`` as there is only one ``BlockType`` instance
+    for every block, however it is recommended to use the ``equals()``.
 
-    For every possible type of block, there is only one ``BlockType`` instance. Therefore, simple comparison with ``==``
-    can be used instead of using the ``equals()`` function.
+Block Data Manipulators
+~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``Location`` class also provides means to check for environmental influences on the block (for example lighting and
-redstone power). The following example checks if a block is a regular chest directly or indirectly powered with redstone.
+The data of a block is held as a ``DataManipulator``, as similar with other parts of the API. This is the container that holds information on components of our block such as the orientation of a block, specific types (stone vs. granite), and so on. Checking the values of these manipulators is easy, you just need to check the block's direction ``DirectionalData``.
 
 .. code-block:: java
 
-    public boolean isPoweredChest(Location block) {
-        if (block.getBlock().getType() == BlockTypes.CHEST) {
-            return block.isBlockPowered() || block.isBlockIndirectlyPowered();
-        } else {
+    import org.spongepowered.api.data.key.Keys;
+    import org.spongepowered.api.data.manipulator.mutable.block.DirectionalData;
+
+    public boolean isFacingNorth(Location<World> blockLoc) {
+        Optional<DirectionalData> optionalData = blockLoc.get(DirectionalData.class);
+        if (!optionalData.isPresent()) {
             return false;
         }
+        DirectionalData data = optionalData.get();
+        if (data.get(Keys.DIRECTION).get().equals(Direction.NORTH)) {
+            return true;
+        }
+        return false;
     }
 
+First, we need to know which ``DataManipulator`` sub-interface we need. Those that are applicable to blocks are found in
+the ``org.spongepowered.api.data.manipulator.mutable`` and ``org.spongepowered.api.data.manipulator.mutable.block`` packages. Then we
+can just pass that class to the ``get()`` method on ``Location`` which will return an ``Optional``. We then have to check if our ``DataManipulator`` actually exists for our block by checking ``ifPresent()``. If it exists, then we can use it. 
+
+More on ``DataManipulator`` s can be found in the :doc:`data documentation <../data/datamanipulators>`.
+
+.. tip ::
+    
+    If a block is guaranteed to have a certain ``DataManipulator`` such as ``DirectionalData`` with stairs, then there is no need to check for ``isPresent()``. Just remove the optional around the ``DataManipulator`` and fetch the non-optional data by adding ``.get()`` to the end of the statement.
```

Comment:
Note: This will cause a Null Pointer Exception if the block ever stops supporting the data. 

If this is an unrecoverable error, we should report it to the user. If the plugin can mostly continue working, ignore it and move on.